wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubUrl.java

```diff
@@ -151,13 +151,28 @@ public String factoryFileLocation() {
   }
 
   /**
-   * Provides map of configured devfile filenames and location to theirs raw content
+   * Provides list of configured devfile filenames with locations
    *
-   * @return map of devfile filenames and locations
+   * @return list of devfile filenames and locations
    */
   @Override
-  public Map<String, String> devfileFileLocations() {
-    return devfileFilenames.stream().collect(Collectors.toMap(key -> key, this::rawFileLocation));
+  public List<DevfileLocation> devfileFileLocations() {
+    return devfileFilenames
+        .stream()
+        .map(
```

Comment:
I'd move the `DevfileLocation` creation into a method, so you can reference the method in the stream.

Reply:
makes sense